test/unit/general/https-support-test.js

```diff
@@ -0,0 +1,263 @@
+ * modify it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * fiware-iotagent-lib is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with fiware-iotagent-lib.
+ * If not, seehttp://www.gnu.org/licenses/.
+ *
+ * For those usages not covered by the GNU Affero General Public License
+ * please contact with::[contacto@tid.es]
+ */
+'use strict';
+
+/* jshint camelcase: false */
+
+var iotAgentLib = require('../../../lib/fiware-iotagent-lib'),
+    request = require('request'),
+    nock = require('nock'),
+    utils = require('../../tools/utils'),
+    groupRegistryMemory = require('../../../lib/services/groups/groupRegistryMemory'),
+    should = require('should'),
+    iotAgentConfig = {
+        logLevel: 'FATAL',
+        contextBroker: {
+            url: 'https://192.168.1.1:1026'
+        },
+        server: {
+            port: 4041
+        },
+        types: {
+            'Light': {
+                commands: [],
+                lazy: [
+                    {
+                        name: 'temperature',
+                        type: 'centigrades'
+                    }
+                ],
+                active: [
+                    {
+                        name: 'pressure',
+                        type: 'Hgmm'
+                    }
+                ],
+                service: 'smartGondor',
+                subservice: 'gardens'
+            },
+            'Termometer': {
+                commands: [],
+                lazy: [
+                    {
+                        name: 'temp',
+                        type: 'kelvin'
+                    }
+                ],
+                active: [
+                ],
+                service: 'smartGondor',
+                subservice: 'gardens'
+            }
+        },
+        service: 'smartGondor',
+        subservice: 'gardens',
+        providerUrl: 'http://smartGondor.com',
+        deviceRegistrationDuration: 'P1M',
+        throttling: 'PT5S',
+        iotManager: {
+            url: 'https://mockediotam.com:9876',
+            path: '/protocols',
+            protocol: 'GENERIC_PROTOCOL',
+            description: 'A generic protocol',
+            agentPath: '/iot'
+        },
+        defaultResource: '/iot/d'
+    },
+    groupCreation = {
+        service: 'theService',
+        subservice: 'theSubService',
+        resource: '/deviceTest',
+        apikey: '801230BJKL23Y9090DSFL123HJK09H324HV8732',
+        type: 'SensorMachine',
+        trust: '8970A9078A803H3BL98PINEQRW8342HBAMS',
+        cbHost: 'https://192.168.1.1:1026',
```

Comment:
If the idea is to illustrate the case of a different CB from the general one used by IOTA (ie. iotAgentConfig.contextBroker) then I'd suggest to use a different URL e.g: `https://192.168.1.1:1027`.

Reply:
not really :) i will just remove it

Reply:
fixed by a7cb7a0

Reply:
If `cbHost` can be removed without effect I think I have understand the feature implemented in this PR completly wrong :)

I mean, the PR is about `device.cbHost` processing. I (wrongly) understand that the `cbHost` you have removed map in some way or another to `device.cbHost`. Thus... how the specific functionality of this PR plays a role in the unit tests? I think I need some guidance...

Reply:
we have three tests (focusing only on the specific tasks)
* register to IoT Manager using HTTPS (no other call is made during this
test) - group registration is used there (so cbHost is not actually used)
* register a new device (without IoT Manager enabled) and see that correct
HTTPS requests are made to context broker: `/NGSI9/registerContext` and
`/v1/updateContext` are used
* subscribe to context broker via HTTPS: `/v1/subscribeContext` and
`/v1/updateContext` are used

Reply:
Ok, great. NTC